### ISSUE TITLE
Relay access denied - (Watchguard?)

### ISSUE THREAD

**Issue by Yodavish** [PHPMailer/PHPMailer#1322]
# Problem description
I'm still new this, so please pardon my lack of knowledge of PHP/phpmailer. I'm able to send a **outbound** email from my local machine, however, when I placed the code on the test server, I receive a "relay access denied". My original code used port '25' with SMTPSecure = 'none' which worked fine on my local machine.  I've tried using SMTPSecure= 'tls' with port 587 and 'ssl' with port = '465' but both resulting in "Connection timeout (110)". The test server is hosted by AWS. Another issue is "WatchGaurd" showing up on the debug output, only when using port '25' and SMTPSecure = 'none'. If someone could point me in the right direction it would be much appreciated.

# Code to reproduce
$email =  $_SESSION['Account']['Email'];
$emailContact = $_SESSION['Account']['Contact Name'];

$BodyConfirmation = 'Thank you for your order!';
$mail = new PHPMailer(); 

$mail->SMTPDebug = 3;                           // Enable verbose debug output
$mail->isSMTP();                                        // Set mailer to use SMTP
$mail->Host = 'smtpmail.propath.com';    // Specify main and backup SMTP servers
$mail->SMTPAuth = false;                         // Enable SMTP authentication
$mail->Username = 'myUserName';         // SMTP username
$mail->Password = 'myPassword';           // SMTP password
$mail->SMTPSecure = 'none';                   // Enable TLS encryption, `ssl` also accepted
$mail->Port = 25;                                      // TCP port to connect to
$mail->SMTPOptions = array(
			'ssl' => array(
				'verify_peer' => false,
				'verify_peer_name' => false,
				'allow_self_signed' => false
)
);
 $mail->setFrom('DoNotReply@propath.com', 'DoNotReply');
$mail->addAddress($email, $emailContact );              // Add a recipient

//Content 
$mail->isHTML(true);                                    // Set email format to HTML
$mail->Subject = $subject;
$mail->Body    = $BodyConfirmation 
		
 if(!$mail->Send()){
    echo "Mailer Error: " . $mail->ErrorInfo;
}
else{
    echo "Message has been sent";
}

# Debug output
2018-01-08 14:50:54 Connection: opening to smtpmail.propath.com:25, timeout=300, options=array ( 'ssl' => array ( 'verify_peer' => false, 'verify_peer_name' => false, 'allow_self_signed' => false, ),)
2018-01-08 14:50:54 Connection: opened
2018-01-08 14:50:54 SERVER -> CLIENT: 220 *****************
2018-01-08 14:50:54 CLIENT -> SERVER: EHLO 52.40.160.248
2018-01-08 14:50:54 SERVER -> CLIENT: 250-watchguard.propath.com250-PIPELINING250-SIZE 131000000250-ETRN250-STARTTLS250-ENHANCEDSTATUSCODES250 8BITMIME
2018-01-08 14:50:54 CLIENT -> SERVER: STARTTLS
2018-01-08 14:50:54 SERVER -> CLIENT: 220 2.0.0 Ready to start TLS
2018-01-08 14:50:55 CLIENT -> SERVER: EHLO 52.40.160.248
2018-01-08 14:50:55 SERVER -> CLIENT: 250-watchguard.propath.com250-PIPELINING250-SIZE 131000000250-ETRN250-ENHANCEDSTATUSCODES250 8BITMIME
2018-01-08 14:50:55 CLIENT -> SERVER: MAIL FROM:<DoNotReply@propath.com>
2018-01-08 14:50:55 SERVER -> CLIENT: 250 2.1.0 Ok
2018-01-08 14:50:55 CLIENT -> SERVER: RCPT TO:<tranvu52@gmail.com>
2018-01-08 14:50:55 SERVER -> CLIENT: 554 5.7.1 <tranvu52@gmail.com>: Relay access denied
2018-01-08 14:50:55 SMTP ERROR: RCPT TO command failed: 554 5.7.1 <tranvu52@gmail.com>: Relay access denied
2018-01-08 14:50:55 CLIENT -> SERVER: QUIT
2018-01-08 14:50:55 SERVER -> CLIENT: 221 2.0.0 Bye
2018-01-08 14:50:55 Connection: closed
SMTP Error: The following recipients failed: tranvu52@gmail.com: <tranvu52@gmail.com>: Relay access denied

#Debug output (When using port 587)
2018-01-08 15:19:16 Connection: opening to smtpmail.propath.com:587, timeout=300, options=array ( 'ssl' => array ( 'verify_peer' => false, 'verify_peer_name' => false, 'allow_self_signed' => false, ),)
2018-01-08 15:21:23 Connection failed. Error #2: stream_socket_client(): unable to connect to smtpmail.propath.com:587 (Connection timed out) [/opt/bitnami/apps/joomla/htdocs/PHPMailer/vendor/phpmailer/phpmailer/src/SMTP.php line 325]
2018-01-08 15:21:23 SMTP ERROR: Failed to connect to server: Connection timed out (110)

#Debug output (When using port 465)
2018-01-08 15:23:01 Connection: opening to ssl://smtpmail.propath.com:465, timeout=300, options=array ( 'ssl' => array ( 'verify_peer' => false, 'verify_peer_name' => false, 'allow_self_signed' => false, ),)
2018-01-08 15:25:08 Connection failed. Error #2: stream_socket_client(): unable to connect to ssl://smtpmail.propath.com:465 (Connection timed out) [/opt/bitnami/apps/joomla/htdocs/PHPMailer/vendor/phpmailer/phpmailer/src/SMTP.php line 325]
2018-01-08 15:25:08 SMTP ERROR: Failed to connect to server: Connection timed out (110)

**Comment by Synchro** [comment]
In the first debug output, you're trying to send without authenticating, and it looks like your server requires that, so set `SMTPAuth = true`. Your `Host` property says `smtpmail.propath.com`, but the server you're connecting to says it's called `watchguard.propath.com`, and the certificate is in that name too, so verification will fail. To fix that I suggest setting `Host = 'watchguard.propath.com'` and removing the verification suppression in `SMTPOptions` - that's never a good thing to have. Note that it's using SMTP+STARTTLS encryption (`SMTPSecure = 'tls'`) because PHPMailer spotted that your server supports encryption and so enables it automatically, but it will fail by default because of the hostname mismatch.

As for the second and third sets of debug output, that just looks like outbound SMTP is blocked from your AWS server, which is common. The tests in the troubleshooting guide will help you confirm that.

**Comment by Yodavish** [comment]
Thanks for the reply.  I've tried updating the code as you suggested but it fails to connect

#Code to reproduce
//Server settings
			$mail->SMTPDebug = 3;                                  // Enable verbose debug output
			$mail->isSMTP();                                       // Set mailer to use SMTP
			$mail->Host = 'watchguard.propath.com'; 	   			   // Specify main and backup SMTP servers
			$mail->SMTPAuth = true;                               // Enable SMTP authentication
			$mail->Username = 'myUserName';                        // SMTP username
			$mail->Password = 'myPassWord';                    // SMTP password
			$mail->SMTPSecure = 'tls';                             // Enable TLS encryption, `ssl` also accepted
			$mail->Port = 587; 


#Debug output
2018-01-08 16:54:46 Connection: opening to watchguard.propath.com:587, timeout=300, options=array()
2018-01-08 16:56:53 Connection failed. Error #2: stream_socket_client(): unable to connect to watchguard.propath.com:587 (Connection timed out) [/opt/bitnami/apps/joomla/htdocs/PHPMailer/vendor/phpmailer/phpmailer/src/SMTP.php line 325]
2018-01-08 16:56:53 SMTP ERROR: Failed to connect to server: Connection timed out (110)

I've tested the telnet on Watchguard with port 587 which fails to connect, "Could not open connection to the host, on port 587: connect failed". However, it does connect using port 25. I've also checked and confirmed that OpenSSL is enabled. I suppose the culprit is AWS blocking the outbound email. I'll see how to unblock it from their side, thanks!

**Comment by Synchro** [comment]
You showed before that port 587 doesn't work, however, SMTP+STARTTLS works fine on port 25 (one of its advantages over SMTPS on port 465), and you've already shown that your TLS config is working (it connected successfully after you suppressed verification) - with the hostname change and removing the suppression it should be all good. I hope you work out what AWS are doing!
